src/Libraries/Microsoft.Extensions.Http.Resilience/Hedging/ResilienceHttpClientBuilderExtensions.Hedging.cs

```diff
@@ -88,26 +90,49 @@ public static IStandardHedgingHandlerBuilder AddStandardHedgingHandler(this IHtt
                     Throw.InvalidOperationException("Request message snapshot is not attached to the resilience context.");
                 }
 
-                var requestMessage = snapshot.CreateRequestMessage();
-
-                // The secondary request message should use the action resilience context
-                requestMessage.SetResilienceContext(args.ActionContext);
-
-                // replace the request message
-                args.ActionContext.Properties.Set(ResilienceKeys.RequestMessage, requestMessage);
-
+                // if a routing strategy has been configured, get the next route from the routing strategy
+                Uri? route;
                 if (args.PrimaryContext.Properties.TryGetValue(ResilienceKeys.RoutingStrategy, out var routingPipeline))
                 {
-                    if (!routingPipeline.TryGetNextRoute(out var route))
+                    // if a routing strategy has been configured but it does not return the next route, then no more routes
+                    // are availabe, stop hedging
+                    if (!routingPipeline.TryGetNextRoute(out route))
                     {
-                        // no routes left, stop hedging
                         return null;
                     }
-
-                    requestMessage.RequestUri = requestMessage.RequestUri!.ReplaceHost(route);
                 }
+                else
+                {
+                    route = null;
+                }
+
+                return async () =>
+                {
+                    Outcome<HttpResponseMessage>? actionResult = null;
+
+                    try
+                    {
+                        var requestMessage = await snapshot.CreateRequestMessageAsync().ConfigureAwait(args.ActionContext.ContinueOnCapturedContext);
+
+                        // The secondary request message should use the action resilience context
+                        requestMessage.SetResilienceContext(args.ActionContext);
+
+                        // replace the request message
+                        args.ActionContext.Properties.Set(ResilienceKeys.RequestMessage, requestMessage);
```

Comment:
```suggestion
                        args.ActionContext.Properties.Set(ResilienceKeys.RequestMessage, requestMessage);
```

There is a breaking change here :/

The `OnHedging` callback does not have access to request message anymore. This is because OnHedging is called after the action is created and before it is invoked.

https://github.com/App-vNext/Polly/blob/f85029c6d14ad20fd36e4fcdde7a32f33409137a/src/Polly.Core/Hedging/Controller/TaskExecution.cs#L127